include/envoy/network/listener.h

```diff
@@ -114,6 +114,51 @@ class ListenerCallbacks {
   virtual void onNewConnection(ConnectionPtr&& new_connection) PURE;
 };
 
+/**
+ * Utility struct that encapsulates the information from a udp socket's
+ * recvfrom/recvmmdg call.
+ *
+ * TODO (conqerAtapple): Maybe this belongs inside the UdpListenerCallbacks
+ * class.
+ */
+struct UdpData {
+  Address::InstanceConstSharedPtr local_address_;
+  Address::InstanceConstSharedPtr peer_address_;
```

Comment:
good point . Will change it to unique_ptr

Reply:
@alyssawilk I was looking at it again and it looks like it might be difficult to change `local_address` to a unique_ptr as it is derived from the  listener's `Socket` object. So I am thinking we could just have the `peer_address` as unique_ptr. It will convey the right ownership semantics but will look odd. WDYT?

Reply:
Wanted to clarify that I was talking about `local_address` that is from the listening socket. `peer_address` could be unique for each packet as its derived from the UDP packet. I assume you mean the same but just wanted to be clear :)

Reply:
@alyssawilk I have updated comments based on ☝️